src/runtime/thread_pool.cc

```diff
@@ -394,9 +397,43 @@ int TVMBackendParallelLaunch(
     FTVMParallelLambda flambda,
     void* cdata,
     int num_task) {
+#ifndef USE_OMP
   int res = tvm::runtime::ThreadPool::ThreadLocal()->Launch(
       flambda, cdata, num_task, 1);
   return res;
+#else
+  const char *val = getenv("TVM_NUM_THREADS");
+  int num_workers;
+  if (val == nullptr) {
+    val = getenv("OMP_NUM_THREADS");
+  }
+  if (val != nullptr) {
+    num_workers = atoi(val);
+  } else {
+#if defined(_M_X64) || defined(__x86_64__)
+  // Half to not count hyper threading.
+    num_workers = std::thread::hardware_concurrency() / 2;
+#else
+    num_workers = std::thread::hardware_concurrency();
+#endif
+  }
+  num_workers = std::max(num_workers, 1);
+  if (num_task ==0) num_task = num_workers;
+  omp_set_num_threads(num_workers);
+  #pragma omp parallel num_threads(num_workers)
+  {
+    TVMParallelGroupEnv env;
+    env.num_task = num_task;
+    std::atomic<int32_t>* sync_counter = new std::atomic<int>[num_task * tvm::runtime::kSyncStride];
+    for (int i = 0; i < num_task; ++i) {
+      sync_counter[i * tvm::runtime::kSyncStride].store(
+          0, std::memory_order_relaxed);
+    }
+    env.sync_handle = sync_counter;
```

Comment:
This part assumes that `need_sync==1`, which is the de facto default now, but we probably need to keep the same logical behavior with the existing thread pool. Another option is to remove `need_sync` from the code. @tqchen Do you want to talk about if we still want `need_sync` in `ThreadPool::Launch()`? https://github.com/dmlc/tvm/blob/master/src/runtime/thread_pool.cc#L298

Reply:
need_sync is used to generate barriers. if we do openmp, perhaps we can ignore it and just use omp's barrier

Reply:
done